src/java/picard/sam/FastqToSam.java

```diff
@@ -135,41 +165,62 @@ public static void main(final String[] argv) {
 
     /* Simply invokes the right method for unpaired or paired data. */
     protected int doWork() {
-            final QualityEncodingDetector detector = new QualityEncodingDetector();
-            final FastqReader reader = new FastqReader(FASTQ,ALLOW_AND_IGNORE_EMPTY_LINES);
-            if (FASTQ2 == null) {
-                detector.add(QualityEncodingDetector.DEFAULT_MAX_RECORDS_TO_ITERATE, reader);
-            } else {
-                final FastqReader reader2 = new FastqReader(FASTQ2,ALLOW_AND_IGNORE_EMPTY_LINES);
-                detector.add(QualityEncodingDetector.DEFAULT_MAX_RECORDS_TO_ITERATE, reader, reader2);
-                reader2.close();
-            }
-            reader.close();
-            
-            QUALITY_FORMAT = detector.generateBestGuess(QualityEncodingDetector.FileContext.FASTQ, QUALITY_FORMAT);
-            if (detector.isDeterminationAmbiguous())
-                LOG.warn("Making ambiguous determination about fastq's quality encoding; more than one format possible based on observed qualities.");
-            LOG.info(String.format("Auto-detected quality format as: %s.", QUALITY_FORMAT));
+        IOUtil.assertFileIsReadable(FASTQ);
+        IOUtil.assertFileIsWritable(OUTPUT);
+
+        FastqReader reader = fileToFastqReader(FASTQ);
+
+        FastqReader reader2 = null;
+        if (FASTQ2 != null) {
+            IOUtil.assertFileIsReadable(FASTQ2);
+            reader2 = fileToFastqReader(FASTQ2);
+        }
+
+        QUALITY_FORMAT = FastqToSam.determineQualityFormat(reader, reader2, QUALITY_FORMAT);
+
+        final SAMFileHeader header = createSamFileHeader();
+        final SAMFileWriter writer = new SAMFileWriterFactory().makeSAMOrBAMWriter(header, false, OUTPUT);
+
+        reader = fileToFastqReader(FASTQ);
+        if (FASTQ2 != null) {
+            reader2 = fileToFastqReader(FASTQ2);
+        }
```

Comment:
I agree, which was part of my problem in the first place as I was sucking the data through on a single named pipe. You could potentially tee the input or something like that but admittedly I was trying to do the minimum amount of change to enable what I needed